File: pepper-farm/backend/schemas/worker_dashboard.py
from pydantic import BaseModel
from typing import Optional


class WorkerAnalyticsResponse(BaseModel):
    openTasksCount: int
    completedTasksCount: int
    avgCompletionTimeHours: Optional[float]
    fastestCompletionTimeHours: Optional[float]
    slowestCompletionTimeHours: Optional[float]
    fastestTaskTitle: Optional[str]
    slowestTaskTitle: Optional[str]

File: pepper-farm/frontend/src/services/workerDashboard.ts
import { apiFetch } from './apiClient';

export interface WorkerAnalytics {
  openTasksCount: number;
  completedTasksCount: number;
  avgCompletionTimeHours: number | null;
  fastestCompletionTimeHours: number | null;
  slowestCompletionTimeHours: number | null;
  fastestTaskTitle: string | null;
  slowestTaskTitle: string | null;
}

export async function getWorkerAnalytics(token: string): Promise<WorkerAnalytics> {
  return apiFetch<WorkerAnalytics>('/api/worker/analytics', {
    headers: { Authorization: `Bearer ${token}` },
  });
}
